Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/service/TasksStatisticsServiceTest.kt

```diff
@@ -0,0 +1,123 @@
+        val testObserver = TestObserver<Event<Statistics>>()
+
+        service.getStatisticsEvent().subscribe(testObserver)
+
+        testObserver.assertReceivedOnNext(listOf(
+                Event.loading<Statistics>().updateData(emptyStatistics())
+        ))
+    }
+
+    @Test
+    fun given_TasksEventIsErrorAndEmpty_on_getStatisticsEvent_it_ShouldSendAnErrorEventWithEmptyStatistics() {
+        tasksEventSubject.onNext(Event.error(SyncError()))
+        val testObserver = TestObserver<Event<Statistics>>()
+
+        service.getStatisticsEvent().subscribe(testObserver)
+
+        testObserver.assertReceivedOnNext(listOf(
+                Event.error<Statistics>(SyncError()).updateData(emptyStatistics())
+        ))
+    }
+
+    @Test
+    fun given_TasksEventIsIdleAndHasTasks_on_getStatisticsEvent_it_ShouldSendAnIdleEventWithStatisticsMatchingTasks() {
+        tasksEventSubject.onNext(Event.idle<Tasks>().updateData(sampleSomeCompletedSomeActiveTasks()))
+        val testObserver = TestObserver<Event<Statistics>>()
+
+        service.getStatisticsEvent().subscribe(testObserver)
+
+        testObserver.assertReceivedOnNext(listOf(
+                Event.idle<Statistics>().updateData(sampleStatisticsSomeCompletedSomeActiveTasks())
+        ))
+    }
+
+    @Test
+    fun given_TasksEventIsLoadingAndHasTasks_on_getStatisticsEvent_it_ShouldSendALoadingEventWithStatisticsMatchingTasks() {
+        tasksEventSubject.onNext(Event.loading<Tasks>().updateData(sampleSomeCompletedSomeActiveTasks()))
+        val testObserver = TestObserver<Event<Statistics>>()
+
+        service.getStatisticsEvent().subscribe(testObserver)
+
+        testObserver.assertReceivedOnNext(listOf(
+                Event.loading<Statistics>().updateData(sampleStatisticsSomeCompletedSomeActiveTasks())
+        ))
+    }
+
+    @Test
+    fun given_TasksEventIsErrorAndHasTasks_on_getStatisticsEvent_it_ShouldSendAnErrorEventWithStatisticsMatchingTasks() {
+        tasksEventSubject.onNext(Event.error<Tasks>(SyncError()).updateData(sampleSomeCompletedSomeActiveTasks()))
+        val testObserver = TestObserver<Event<Statistics>>()
+
+        service.getStatisticsEvent().subscribe(testObserver)
+
+        testObserver.assertReceivedOnNext(listOf(
+                Event.error<Statistics>(SyncError()).updateData(sampleStatisticsSomeCompletedSomeActiveTasks())
+        ))
+    }
+
+    private fun setUpService() {
+        tasksEventSubject = BehaviorSubject.create()
+        val tasksEventReplay = tasksEventSubject.replay()
+        tasksEventReplay.connect()
+        Mockito.`when`(tasksService.getTasksEvent()).thenReturn(tasksEventReplay)
+    }
+
+    private fun emptyStatistics() = Statistics.builder().build()
+
+    private fun sampleStatisticsSomeCompletedSomeActiveTasks() = Statistics.builder()
+            .activeTasks(1)
+            .completedTasks(3)
+            .build()
+
+    private fun sampleSomeCompletedSomeActiveTasks() = Tasks.asSynced(listOf(
```

Comment:
`tasksWithSomeActiveAndSomeCompleted` ? Might just read a little easier, i don't think `sample` is adding much here